modules/storages/app/controllers/storages/admin/health_status_controller.rb

```diff
@@ -48,6 +48,14 @@ def admin_or_frame_layout
 
       def show
         @report = Rails.cache.read(validator.report_cache_key)
+
+        respond_to do |format|
+          format.html
+          format.text do
+            filename = "#{@storage.name.underscore}_health_report_#{@report.latest_timestamp.iso8601}.txt"
```

Comment:
:red_circle: This will fail if the `cache.read` does not return a value (`@report` is nil), which is possible. The HTML version of the page would then render a nice fallback text about regenerating the report, but here we would already fail at generating the file name.

Reply:
FYI: Unresolving the discussion, since I can't see any change/fix to the problem yet.

The most plausible scenario for me that this could occur in the wild is someone having a tab open for many hours, then receiving a response from OP support that they should download a report and then clicking the button for the report that is long vanished in the background.

Reply:
What do you propose then? Rerun the report and send it?

Reply:
That'd be the safest thing to do. If we do it for the entire action, this would mean that you can never see a "no report" page.

The only alternative that I can come up with is to make a fallback for the filename generation that works with a `@report == nil`. The generation of the file content already seems to be fit for it. My preferred way is the automatic regeneration, though.